### PR TITLE
[docs] add new showcase page

### PR DESCRIPTION
**Pull request by sarah11918**
This adds a Showcase page listing Starlight sites in the wild... until we can make a nice-looking one!  😅

### CODE REVIEW

**Review by TheOtterlord — APPROVED** [review]
Looks great :partying_face:
Definitely potential for a fun showcase design later on

File: docs/src/content/docs/showcase.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: Starlight Showcase
+description: See sites built with Starlight around the web!
+---
+Starlight is already being used in production around the web. See some examples below, and PR your own link if you have built your documentation site with Starlight!
+
+- [PubIndexAPI Docs](https://docs.pubindexapi.com/)
```
Comment (sarah11918) [inline docs/src/content/docs/showcase.md:7]:
```suggestion
- [PubIndexAPI Docs](https://docs.pubindexapi.com/)
- [har.fyi](https://har.fyi/)
```

**Review by ElianCodes — APPROVED** [review]
This is very cool! Can't wait to see what @doodlemarks will come up with in the end!

File: docs/src/content/docs/showcase.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+- [har.fyi](https://har.fyi/)
+- [QBCore Docs](https://brycerussell.github.io/qbcore-docs/)
+- [Astro Error Pages](https://astro-error-page-documentation.vercel.app/)
+- [MES Docs](https://docs.muslimeng.org/)
```
Comment (delucis) [inline docs/src/content/docs/showcase.md:11]: This one seems a bit WIP, so suggesting replacing with these two that are more polished:

```suggestion
- [Felicity](https://felicity.pages.dev/)
- [NgxEditor](https://sibiraj-s.github.io/ngx-editor/)
```

Going to suggest holding off merging this until we have #188 ready.

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

**Pull request merged**